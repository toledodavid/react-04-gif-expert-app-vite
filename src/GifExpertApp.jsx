import { useState } from 'react';

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['Vikings', 'Dragon ball']);

  const onAddCategory = () => {
    setCategories(['Valorant', ...categories]);

    // SAME BUT USING THE CALLBACK FROM THE setCategories
    // setCategories(categories => [...categories, 'Valorant']);
  };

  return(
    <>
      <h1>GifExpertApp</h1>

      {/* Input */}

      <button onClick={onAddCategory}>Add</button>

      <ol>
        {
          categories.map((category, index) => (
            <li key={index}>{category}</li>
          ))
        }
      </ol>
    </>
  );
}