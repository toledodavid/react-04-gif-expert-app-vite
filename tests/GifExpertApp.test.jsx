import { fireEvent, render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Tests for GifExpertApp', () => {

  test('should match with the snapshot', () => {
    const {container} = render(<GifExpertApp />);

    expect(container).toMatchSnapshot();
  });

  test('should add a new category to its state', () => {
    const inputValue = 'Nintendo';

    render(<GifExpertApp />);

    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form');

    fireEvent.input(input, {target: {value: inputValue}});
    fireEvent.submit(form);

    expect(input.value).toBeFalsy();
    expect(screen.getAllByRole('heading', {level: 3}).length).toBe(2);
  });

});