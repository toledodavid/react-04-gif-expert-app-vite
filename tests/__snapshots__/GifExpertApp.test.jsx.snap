// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifExpertApp should match with the snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Vikings
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
